dirty_cat/_datetime_encoder.py

```diff
@@ -218,7 +218,7 @@ def transform(self, X, y=None) -> np.ndarray:
 
         Returns
         -------
-        np.ndarray, shape (n_samples, n_features_out_)
+        np.ndarray, shape (n_samples, n_features_out\\_)
```

Comment:
Oh no, that's horrible. I fixes the warning but makes the docstring unreadable in plain text.

Can we try something like `n_features_out_` rather? (I don't know if it fixes the warning)